girder/models/file.py

```diff
@@ -182,14 +183,26 @@ def createLinkFile(self, name, parent, parentType, url, creator, size=None, mime
         :type size: int
         :param mimeType: The mimeType of the file. (optional)
         :type mimeType: str
+        :param reuseExisting: If a file with the same name already exists in
+            this location, return it rather than creating a new file.
+        :type reuseExisting: bool
         """
+
         if parentType == 'folder':
             # Create a new item with the name of the file.
             item = self.model('item').createItem(
-                name=name, creator=creator, folder=parent)
+                name=name, creator=creator, folder=parent, reuseExisting=reuseExisting)
         elif parentType == 'item':
             item = parent
 
+        if reuseExisting:
+            existing = self.findOne({
+                'itemId': item['_id'],
+                'name': name
+            })
+            if existing:
+                return existing
```

Comment:
For a regular file, creation of the file document and updating its contents are independent, so there isn't a parallel, but I would imagine that if we had a `reuseExisting` flag on the upload process, it would update the contents and use the existing file document. In this case, since we have the url already, I think it makes the most sense to update it since the user is requesting that to be the URL.